File: VB/MyBarManager/MySkinBarManagerPaintStyle.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraBars.Styles

Namespace WindowsApplication1
	Public Class MySkinBarManagerPaintStyle
		Inherits SkinBarManagerPaintStyle
		Public Sub New(ByVal collection As BarManagerPaintStyleCollection)
			MyBase.New(collection)
		End Sub
		Protected Overrides Sub RegisterItemInfo()
			MyBase.RegisterItemInfo()
			Dim list As BarItemInfo = ItemInfoCollection(MyBarButtonItem.BaseBarItemName)
			ItemInfoCollection.Add(New BarItemInfo(MyBarButtonItem.BarItemName, MyBarButtonItem.BarItemCaption, -1, GetType(MyBarButtonItem), list.LinkType, list.ViewInfoType, list.LinkPainter, True, False))
		End Sub
	End Class
End Namespace

File: CS/MyBarManager/MySkinBarManagerPaintStyle.cs
using System;
using DevExpress.XtraBars.Styles;

namespace WindowsApplication1 {
    public class MySkinBarManagerPaintStyle : SkinBarManagerPaintStyle {
        public MySkinBarManagerPaintStyle(BarManagerPaintStyleCollection collection) : base(collection) { }
        protected override void RegisterItemInfo() {
            base.RegisterItemInfo();
            BarItemInfo list = ItemInfoCollection[MyBarButtonItem.BaseBarItemName];
            ItemInfoCollection.Add(new BarItemInfo(MyBarButtonItem.BarItemName, MyBarButtonItem.BarItemCaption, -1, typeof(MyBarButtonItem), list.LinkType, list.ViewInfoType, list.LinkPainter, true, false));
        }
    }
}
